src/bootstrap/dist.rs

```diff
@@ -1890,6 +1890,15 @@ fn maybe_install_llvm(builder: &Builder<'_>, target: TargetSelection, dst_libdir
         }
     }
 
+    let copy = |src: &Path, dstdir, perms| {
+        // Try to use a hard link for this file, but only if it's not already a symbolic link
+        if src.is_symlink() {
+            builder.install(src, dstdir, perms)
+        } else {
+            builder.copy(src, &dstdir.join(src.file_name().unwrap()))
+        }
+    };
```

Comment:
Going to r+ this for now, since I think it's OK, but we should think about whether we can make copy() handle symlinks by operating on the underlying inode (whether hardlink or raw byte copy). I'm not a fan of peeking into the internals of install/copy like this, so if we do this in more places we should try to move away from that.

Reply:
copy already handles symlinks, it just does so by copying the symlink itself instead of dereferencing it and making a hard link to the target. I'm not sure why that behavior was chosen; I suppose I could try changing it, but then we'd no longer set permissions on the copied file.

Umm actually I think that's a new bug with this change, it won't set permissions for files which aren't symlinks.